src/api/world.py

```diff
@@ -24,16 +25,24 @@ class WorldFormat:
     Base class for World objects
     """
 
-    mapping_handler: numpy.ndarray = None
+    mapping_handler: BlockManager = None
     _materials = None
 
+    def __init__(self, adapters: Dict[str, Callable[[Any], Any]]):
```

Comment:
Maybe instead of an "adapters" dictionary parameter, each kind of adapter will be its own parameter?  For example. here we'll have a "get_blockstate_adapter"
```suggestion
    def __init__(self, get_blockstate_adapter: Callable[[str], Block]):
```

Reply:
Note: Don't commit/apply that "suggested change" as-is since it requires changes in other places too

Reply:
I don't really agree, since if we add in more adapters for things like entities/block entities/future features, we'll have to keep adding parameters vs. just adding them to a dictionary. I was think that we have default implementations of each adapter in WorldFormat, then if needed a version can override none/one/more of the default implementations

Reply:
I agree with the fact that we'll add a bunch more and that we'll have default behaviors for each of them and I understand your concern of it being too many parameters, but I think such an important part of the code be explicit, even it means the function will have a lot of parameters to it.